Sprint-3/todo-list/script.js

```diff
@@ -1,25 +1,97 @@
+const toDoInput = document.getElementById("inputToDo");
+const addToDoBtn = document.getElementById("addToDo");
+const container = document.getElementById("todo-list");
+const removeAll = document.getElementById("remove-all-completed");
+addToDoBtn.addEventListener("click", () => {
+  addNewTodo(event);
+});
+let todos = [];
+
+// Function to populate the todo list
 function populateTodoList(todos) {
   let list = document.getElementById("todo-list");
-  // Write your code to create todo list elements with completed and delete buttons here, all todos should display inside the "todo-list" element.
+  list.innerHTML = ""; // Clear the list to avoid duplications
+
+  todos.forEach((todo, index) => {
+    const listItem = document.createElement("li");
+    listItem.innerText = todo.task;
+    list.appendChild(listItem);
+
+    // Create 'Complete' button
+    const completeBtn = document.createElement("button");
+    completeBtn.innerText = "complete";
+    const newButton1 = listItem.appendChild(completeBtn);
+    newButton1.className = "newButton";
+    completeBtn.addEventListener("click", () => {
+      todos[index].complete = true;
+      populateTodoList(todos);
+    });
+    if (todo.complete) {
+      listItem.style.textDecoration = "line-through";
+    }
+    // Create 'Delete' button
+    const deleteBtn = document.createElement("button");
+    deleteBtn.innerText = "delete";
+    const newButton = listItem.appendChild(deleteBtn);
+    newButton.className = "newButton";
+    deleteBtn.addEventListener("click", () => {
+      todos.splice(index, 1); // Remove the todo from the array
+      populateTodoList(todos); // Refresh the list
+    });
+
+    // Apply completed style if the todo is already marked as complete
+  });
+}
+
+// Function to add a new todo
+function addNewTodo(event) {
```

Comment:
Why are there two definition of `addNewTodo(event)` in your code?

Reply:
I removed one of them,
thanks CJ